src/Search.php

```diff
@@ -5165,7 +5165,7 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
             case "glpi_ticketvalidations.status":
             case "glpi_changes.global_validation":
             case "glpi_changevalidations.status":
-                if ($val == 'all') {
+                if ($val == 'all' || !is_numeric($val)) {
```

Comment:
Maybe with a `&&` instead of a `||`. I guess we need a test here.
```suggestion
                if (!in_array($val, ['all', 'can'], true) && !is_numeric($val)) {
```

Reply:
Indeed